presto-main/src/main/java/io/prestosql/server/PingResource.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server;
+
+import javax.ws.rs.HEAD;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+
+import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
+
+@Path("/")
```

Comment:
Curious - is that possible that `HeartbeatFailureDetector` pings some already exitent, known url? Instead of adding a fake one?

Reply:
I did some digging but there wasn't a clear better endpoint to ping. The goal was to keep the endpoint as cheap as possible for both the client and server, hoping it could also be used to serve other "liveness check" purposes.

Do you have an alternative endpoint in mind?

Reply:
No. Not really. Just using "/" seems kinda weird. Maybe we should add a specific endpoint "/ping" for pinging purposes?

Or am I reading it wrong and it already serves on "/ping" and the path is derived from method name?

Reply:
You would be correct that this is the root path (eg: `http://localhost:8080/`) since that's currently where the hearbeat pings are directed. I'm not 100% confident in how the service discovery system wires things together so I wasn't sure how to go about changing the endpoint used there is registered and instead put this handler on the path currently used. Would it be as easy as making this `@Path("/ping")` and putting a hard-coded "/ping" suffix [here](https://github.com/prestosql/presto/blob/fca01af7715afd531261343b43dd88d3642e56d6/presto-main/src/main/java/io/prestosql/failuredetector/HeartbeatFailureDetector.java#L276)?

Reply:
Don't know neither. @electrum?

Reply:
The server does a redirect to `/ui` when you hit `/`. Not sure why you're seeing 404. Regardless, binding a resource to `/` is problematic because that url is handled by some other part of Presto, and HEAD should be consistent with GET (minus the response body).

Reply:
Only the coordinator has that redirect installed, workers return 404. I'm fine with moving the route to `/ping` if that's where you think it should go. The question then becomes whether / how to relate the ping endpoint through `ServiceDescriptor` since this is really a per-node endpoint.

Reply:
Ah, I guess that broke with the recent UI changes. 

The URI in the service descriptor is published automatically via:

```
        discoveryBinder(binder).bindHttpAnnouncement("presto")
                .addProperty("node_version", nodeVersion.toString())
                .addProperty("coordinator", String.valueOf(serverConfig.isCoordinator()));
```

If we wanted to have a separate URI for ping, we'd need to either:
* add a property to the announcement to indicate what's the (relative) URI for that endpoint
* hard-code the URI in the failure detector

Both have pros and cons -- let me think about this a little more.

Reply:
I think it’s reasonable to add a well known `/ping` endpoint that’s guaranteed to exist on the coordinator and workers and is accessible without authentication. This is useful for external monitoring, load balancers, etc.

Reply:
We already have such an endpoint. It's called `/v1/status`

Reply:
That's a little bit heavy for constant background pinging though, right? We could, I suppose, move this into the `StatusResource` and handle HEAD requests specifically. Would that work with everyone?

Reply:
What it does is actually pretty cheap. The failure detector calls it every half a second anyway, so the impact is minimal. However, adding handling for HEAD in StatusResource is a good idea.